django/utils/inspect.py

```diff
@@ -1,8 +1,14 @@
+import functools
 import inspect
 
 
+@functools.lru_cache(None)
+def get_signature(func):
+    return inspect.signature(func)
```

Comment:
I initially thought it would be an issue for `boundmethod` instances but it looks like they hash to the same value even if they are bound to different instances

```python
class Foo:
    def __init__(self):
        pass

>>> Foo().__init__ == Foo().__init__
False
>>> hash(Foo().__init__) == hash(Foo().__init__)
True
``` 

In all cases I don't see a compelling reason for setting `maxsize=None` and allowing the cache to grow unbounded.

Reply:
Thanks for pointing that out. Setting `maxsize=None` was out of pure habit since I normally use lru_cache to memoize my top-down dynamic programming functions.  

How does setting `max_size` to the default value of 128 sound? Thoughts?